app/src/main/java/com/infomaniak/mail/workers/DraftsActionsWorker.kt

```diff
@@ -267,6 +267,7 @@ class DraftsActionsWorker @AssistedInject constructor(
 
     private fun Draft.uploadAttachments(): Result {
         val attachmentsToUpload = getNotUploadedAttachments(draft = this)
+        val attachmentsNames = attachmentsToUpload.map { it.name }.toMutableList()
```

Comment:
Don't loop here, do it directly in `getNotUploadedAttachments()` to save a loop.

Reply:
Done in the next PR